sdk/samples/iot/paho_iot_pnp_with_provisioning_sample.c

```diff
@@ -0,0 +1,1218 @@
+
+      handle_command_request(message, &command_request);
+    }
+    else
+    {
+      IOT_SAMPLE_LOG_ERROR("Message from unknown topic: az_result return code 0x%08x.", rc);
+      IOT_SAMPLE_LOG_AZ_SPAN("Topic:", topic_span);
+      exit(rc);
+    }
+  }
+}
+
+static void handle_device_property_message(
+    MQTTClient_message const* message,
+    az_iot_pnp_client_property_response const* property_response)
+{
+  az_span const message_span = az_span_create((uint8_t*)message->payload, message->payloadlen);
+
+  // Invoke appropriate action per response type (3 types only).
+  switch (property_response->response_type)
+  {
+    // A response from a property GET publish message with the property document as a payload.
+    case AZ_IOT_PNP_CLIENT_PROPERTY_RESPONSE_TYPE_GET:
+      IOT_SAMPLE_LOG("Message Type: GET");
+      process_device_property_message(message_span, property_response->response_type);
+      break;
+
+    // An update to the desired properties with the properties as a payload.
+    case AZ_IOT_PNP_CLIENT_PROPERTY_RESPONSE_TYPE_DESIRED_PROPERTIES:
+      IOT_SAMPLE_LOG("Message Type: Desired Properties");
+      process_device_property_message(message_span, property_response->response_type);
+      break;
+
+    // A response from a reported properties publish message.
+    case AZ_IOT_PNP_CLIENT_PROPERTY_RESPONSE_TYPE_REPORTED_PROPERTIES:
+      IOT_SAMPLE_LOG("Message Type: Reported Properties");
+      break;
+  }
+}
+
+static void process_device_property_message(
+    az_span message_span,
+    az_iot_pnp_client_property_response_type response_type)
+{
+  az_json_reader jr;
+  az_result rc = az_json_reader_init(&jr, message_span, NULL);
+  IOT_SAMPLE_EXIT_IF_AZ_FAILED(rc, "Could not initialize json reader");
+
+  int32_t version_number;
+  rc = az_iot_pnp_client_property_get_property_version(
+      &pnp_client, &jr, response_type, &version_number);
+  IOT_SAMPLE_EXIT_IF_AZ_FAILED(rc, "Could not get property version");
+
+  rc = az_json_reader_init(&jr, message_span, NULL);
+  IOT_SAMPLE_EXIT_IF_AZ_FAILED(rc, "Could not initialize json reader");
+
+  double desired_temperature;
+  az_span component_name;
+  az_json_reader property_name_and_value;
+
+  while (az_result_succeeded(az_iot_pnp_client_property_get_next_component_property(
+      &pnp_client, &jr, response_type, &component_name, &property_name_and_value)))
+  {
+    if (az_json_token_is_text_equal(
+            &property_name_and_value.token, property_desired_temperature_name))
+    {
+      rc = az_json_reader_next_token(&property_name_and_value);
+      if (az_result_failed(rc))
+      {
+        IOT_SAMPLE_LOG_ERROR("Could not move to property value");
+      }
+
+      rc = az_json_token_get_double(&property_name_and_value.token, &desired_temperature);
+      if (az_result_failed(rc))
+      {
+        IOT_SAMPLE_LOG_ERROR("Could not get property value");
+      }
+
+      IOT_SAMPLE_LOG(" "); // Formatting
+
+      bool confirm = true;
```

Comment:
What are the semantics of this `confirm` boolean?

Reply:
there is a confirmation with status semantic with pnp which tells the service it received a property and lets it know whether it accepts it or not. This is basically a notice to the API that it needs to be wrapped in that confirmation construct.